test/index.js

```diff
@@ -34,14 +33,26 @@ const eq = actual => expected => {
   strictEqual (Z.equals (actual, expected), true);
 };
 
-const testModule = (results, path, options) => {
-  results.forEach (([description, expected], idx) => {
-    test (`${relative ('test', path)} \u001B[2m›\u001B[0m ${description}`, () =>
-      doctest (options) (path)
-      .then (actuals => {
-        eq (actuals[idx]) (expected);
-      })
-    );
+const testModule = (expecteds, path, options) => {
+  let promise = null;
+  const run = () => (promise ?? (promise = doctest (options) (path)));
```

Comment:
This allows test cases to share a promise, but postpones the creation of the promise until the first of these test cases is run.